src/ui/public/management/__tests__/section.js

```diff
@@ -0,0 +1,108 @@
+import expect from 'expect.js';
+
+import ManagementSection from 'ui/management/section';
+import IndexedArray from 'ui/indexed_array';
+
+describe('ManagementSection', () => {
+  describe('constructor', () => {
+    it('defaults display to id', () => {
+      const section = new ManagementSection('kibana');
+      expect(section.display).to.be('kibana');
+    });
+
+    it('exposes items', () => {
+      const section = new ManagementSection('kibana');
+      expect(section.items).to.beEmpty;
+    });
+
+    it('sets url based on path', () => {
+      const section = new ManagementSection('kibana', { path: 'foo' });
+      expect(section.url).to.be('#/management/foo');
+    });
+
+    it('assigns all options', () => {
+      const section = new ManagementSection('kibana', { description: 'test' });
+      expect(section.description).to.be('test');
+    });
+  });
+
+  describe('register', () => {
+    let section;
+
+    beforeEach(() => {
+      section = new ManagementSection('kibana');
+    });
+
+    it('returns a ManagementSection', () => {
+      expect(section.register('about')).to.be.a(ManagementSection);
+    });
+
+    it('provides a reference to the parent', () => {
+      expect(section.register('about').parent).to.eql(section);
+    });
+
+    it('adds item', function () {
+      section.register('about', { description: 'test' });
+
+      expect(section.items).to.have.length(1);
+      expect(section.items[0]).to.be.a(ManagementSection);
+      expect(section.items[0].id).to.be('about');
+    });
+
+    it('can only register a section once', () => {
+      let threwException = false;
+      section.register('about');
+
+      try {
+        section.register('about');
+      } catch (e) {
+        threwException = e.message.indexOf('is already registered') > -1;
+      };
+
+      expect(threwException).to.be(true);
```

Comment:
I actually ran into an issue when initially trying to use withArgs. Here is the open issue for it: https://github.com/Automattic/expect.js/issues/146